mq/inmemory/mq.h

```diff
@@ -2,30 +2,104 @@
 #define BRICKS_MQ_INMEMORY_MQ_H
 
 // MMQ is an efficient in-memory FIFO buffer.
+// One of the objectives of MMQ is to minimize the time for which the message pushing thread is blocked for.
 //
-// Messages can be pushed into it via `PushMessage()`.
+// Messages can be pushed into it via thread-safe methods `PushMessage()` or `EmplaceMessage()`.
 // The consumer is run in a separate thread, and is fed one message at a time via `OnMessage()`.
-// The order of messages is preserved.
 //
-// One of the objectives of MMQ is to minimize the time for which the message pushing thread is blocked for.
+// The buffer size, i.e. the number of the messages MMQ can hold, is defined by
+// the constructor argument `buffer_size`. For usability reasons the default
+// value for it can be set via `DEFAULT_BUFFER_SIZE` template argument.
 //
-// The default (and only so far) implementation discards older messages if the queue gets over capacity.
-// TODO(dkorolev): This behavior should probably change.
+// There are two possible strategies in case of buffer overflow (i.e. there is
+// no free space to store message at the next call to `PushMessage()` or
+// `EmplaceMessage()`):
+//   1) Discard (drop) the message. In this case, the number of the messages dropped between
+//      the subseqent calls of the consumer may be passed as a second argument of `OnMessage()`.
+//   2) Block the pushing thread and wait for the next message to be consumed
+//      and free the space in the buffer.
+//      IMPORTANT NOTE: if there are several threads waiting to push the
+//      message, MMQ DOES NOT guarantee that the messages will be added in the
+//      order in which the functions were called. However, for any particular
+//      thread MMQ DOES GUARANTEE the order of the messages for the subsequent
+//      requests to push the message.
+//  This behavior of MMQ can be controlled via `DROP_ON_OVERFLOW` template
+//  argument.
```

Comment:
Nit: A lot of these comments seem to be wrapped at 80 characters line width, is it intentional?